### PR TITLE
Logic to delete old iocs and add ioc index rollover

### PR DESCRIPTION
**Pull request by jowg-amazon**
### Description
Rollsover ioc indices based on age of index and number of indices associated with an alias. Deletes the old ioc indices
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java [hunk 1]
```diff
@@ -191,4 +191,18 @@ public static final List<Setting<?>> settings() {
         return List.of(BATCH_SIZE, THREAT_INTEL_TIMEOUT, TIF_UPDATE_INTERVAL);
     }
 
+    // Threat Intel IOC Settings
+    public static final Setting<TimeValue> IOC_INDEX_RETENTION_PERIOD = Setting.positiveTimeSetting(
+            "plugins.security_analytics.ioc_history_retention_period",
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:196]: plugins.security_analytics.ioc.history_retention_period

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java [hunk 2]
```diff
@@ -191,4 +191,18 @@ public static final List<Setting<?>> settings() {
...
+            new TimeValue(30, TimeUnit.DAYS),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+    );
+
+    public static final Setting<Integer> MAX_IOC_INDICES_PER_ALIAS = Setting.intSetting(
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:201]: **`plugins.security_analytics.ioc.`** should be prefix

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java [hunk 3]
```diff
@@ -191,4 +191,18 @@ public static final List<Setting<?>> settings() {
         return List.of(BATCH_SIZE, THREAT_INTEL_TIMEOUT, TIF_UPDATE_INTERVAL);
     }
 
+    // Threat Intel IOC Settings
+    public static final Setting<TimeValue> IOC_INDEX_RETENTION_PERIOD = Setting.positiveTimeSetting(
+            "plugins.security_analytics.ioc_history_retention_period",
+            new TimeValue(30, TimeUnit.DAYS),
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:197]: set a min value of 1 day

Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:197]: done

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java (hunk 2 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java]: changed to `ioc_indices_per_alias`

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java (hunk 1 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java]: fixed

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java [hunk 4]
```diff
@@ -191,4 +191,19 @@ public static final List<Setting<?>> settings() {
...
+    // Threat Intel IOC Settings
+    public static final Setting<TimeValue> IOC_INDEX_RETENTION_PERIOD = Setting.timeSetting(
+            "plugins.security_analytics.ioc_index_retention_period",
+            new TimeValue(30, TimeUnit.DAYS),
+            new TimeValue(1, TimeUnit.DAYS),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+    );
+
+    public static final Setting<Integer> IOC_INDICES_PER_ALIAS = Setting.intSetting(
+            "plugins.security_analytics.ioc_indices_per_alias",
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:203]: `plugins.security_analytics.ioc.`

Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java:203]: max_indices*

**Review by eirsep — CHANGES_REQUESTED** [review]

File: src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java [hunk 5]
```diff
@@ -50,11 +52,8 @@ public class STIX2IOCFeedStore implements FeedStore {
     public static final String IOC_ALL_INDEX_PATTERN = IOC_INDEX_NAME_BASE + "-*";
     public static final String IOC_FEED_ID_PLACEHOLDER = "FEED_ID";
     public static final String IOC_INDEX_NAME_TEMPLATE = IOC_INDEX_NAME_BASE + "-" + IOC_FEED_ID_PLACEHOLDER;
-
-    // TODO hurneyt implement history indexes + rollover logic
-    public static final String IOC_HISTORY_WRITE_INDEX_ALIAS = IOC_INDEX_NAME_TEMPLATE + "-history-write";
-    public static final String IOC_HISTORY_INDEX_PATTERN = "<." + IOC_INDEX_NAME_BASE + "-history-{now/d{yyyy.MM.dd.hh.mm.ss|UTC}}-1>";
-
+    public static final String IOC_HISTORY_WRITE_INDEX_ALIAS = IOC_INDEX_NAME_TEMPLATE + "-write";
```
[the remark this answers is not in the document]
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java]: removed history

File: src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java (hunk 4 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java]: fixed

Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/settings/SecurityAnalyticsSettings.java]: fixed to `plugins.security_analytics.ioc.max_indices_per_alias`

**Review by eirsep — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**